### DISCUSSION TITLE
overwritable_footstep_index_offsetが0と1以外の値に対応不可

### DISCUSSION THREAD

**Discussion by Naoki-Hiraoka** [YutaKojio/hrpsys-base#38] [Q&A] [answered]
この部分が常に`false`で`lcg.get_footstep_index()`の値が使われるので、`overwritable_footstep_index_offset`が0と1以外の値の場合に対応できず、
https://github.com/YutaKojio/hrpsys-base/blob/eb90ec0f95072a878a7a0bb9035824e002c7be98/rtc/AutoBalancer/GaitGenerator.cpp#L2250

この行で破綻しそうです。(実際にchoreonoidで転んだりセグフォったりしました)
https://github.com/YutaKojio/hrpsys-base/blob/eb90ec0f95072a878a7a0bb9035824e002c7be98/rtc/AutoBalancer/GaitGenerator.cpp#L775

`overwritable_footstep_index_offset`をわざわざ0と1以外の値にすることはまず無いと思うので、直す優先順位は低いですが...

**Comment by YutaKojio** [comment]
update_vel を昔入れてた時期があって，結局
https://github.com/YutaKojio/hrpsys-base/blob/eb90ec0f95072a878a7a0bb9035824e002c7be98/rtc/AutoBalancer/GaitGenerator.cpp#L775
では消したので，変な状態になっている気がする．後で確認して直します．

確認だけど，

> 実際にchoreonoidで転んだりセグフォったりしました

は，overwritable_footstep_index_offsetが2以上で go-velocity したとき？

**Reply by Naoki-Hiraoka** [reply]
> overwritable_footstep_index_offsetが2以上で go-velocity したとき？

はい。

**Comment by YutaKojio** [comment, marked as answer]
https://github.com/YutaKojio/hrpsys-base/commit/8e725bbe0a97a5a281cd2cf67803cff75a3244ea
で0 ~ 3の場合でも対応できるようにしました．
4以上のときはsetするときにprint文出すようにしています．
